rxmongo/src/main/scala/akka/contrib/persistence/mongodb/RxMongoJournaller.scala

```diff
@@ -159,31 +160,38 @@ class RxMongoJournaller(driver: RxMongoDriver) extends MongoPersistenceJournalli
   }
 
   private[mongodb] override def deleteFrom(persistenceId: String, toSequenceNr: Long)(implicit ec: ExecutionContext) = {
-    val journal = driver.getJournal(persistenceId)
-    val query = journalRangeQuery(persistenceId, 0L, toSequenceNr)
-    val update = BSONDocument(
-      "$pull" -> BSONDocument(
-        EVENTS -> BSONDocument(
-          PROCESSOR_ID -> persistenceId,
-          SEQUENCE_NUMBER -> BSONDocument("$lte" -> toSequenceNr))),
-      "$set" -> BSONDocument(FROM -> (toSequenceNr + 1)))
-    val remove = BSONDocument("$and" ->
-      BSONArray(
-        BSONDocument(PROCESSOR_ID -> persistenceId),
-        BSONDocument(EVENTS -> BSONDocument("$size" -> 0))))
     for {
+      journal <- driver.getJournal(persistenceId)
       ms <- findMaxSequence(persistenceId, toSequenceNr)
-      j <- journal
-      wrUpdate <- j.update(query, update, writeConcern, upsert = false, multi = true)
-      if wrUpdate.ok
       _ <- ms.fold(Future.successful(()))(setMaxSequenceMetadata(persistenceId, _))
-      wr <- j.remove(remove, writeConcern)
+
+
+      //first remove docs that have to be removed, it avoid settings some docs with from > to and trying to set same from on several docs
+      docWithAllEventsToRemove = BSONDocument(PROCESSOR_ID -> persistenceId, TO -> BSONDocument("$lte" -> toSequenceNr))
+      removed <- journal.remove(docWithAllEventsToRemove)
+      if (removed.ok)
+
+
+      //then update the (potential) doc that should have only one (not all) event removed
+      //note the query: we exclude documents that have to < toSequenceNr, it should have been deleted just before,
+      // but we avoid here some potential race condition that would lead to have from > to and several documents with same from
+      query = journalRangeQuery(persistenceId, toSequenceNr, toSequenceNr)
+      update = BSONDocument(
+        "$pull" -> BSONDocument(
+          EVENTS -> BSONDocument(
+            PROCESSOR_ID -> persistenceId,
+            SEQUENCE_NUMBER -> BSONDocument("$lte" -> toSequenceNr))),
+        "$set" -> BSONDocument(FROM -> (toSequenceNr + 1)))
+
+      wrUpdate <- journal.update(query, update, writeConcern, upsert = false, multi = true)
+      if wrUpdate.ok
```

Comment:
I think this looks like a good translation except for the `DuplicateKey` race condition handling.

Reply:
It seems to be no catch for DuplicateKey in setMaxSequenceMetadata either. (#151 fix)
Thus I through it was different for rxmongo.
I will try to have a look

Reply:
Do you know what would be the error (and how to catch it) for a duplicatekey error with rxmongo driver ? (on GenericCollection.update() method)

Reply:
I think there's an error code on the write result object that comes back.

According to `com.mongodb.ErrorCategory` in the java driver, the duplicate key codes are `11000, 11001, 12582`

Reply:
I added a recover to catch the duplicate key error.